### ISSUE TITLE
[Form validation] Checking uniqueness

### ISSUE THREAD

**Issue by IonutBajescu** [Semantic-Org/Semantic-UI#3518]
Hey there,

I'm using the semantic ui validation in one of my angular projects and I noticed that validating the uniqueness of a field is not officially supported.

I can see why it wasn't added so far, given the instantaneously nature of the form validation, calling an external URL will not be possible without refactoring the form module to accept promises from its rules. 

Any future plans of supporting async validation rules? I could help with the implementation if needed.

**Comment by xiechao06** [comment]
Previously, I submit this issue: https://github.com/Semantic-Org/Semantic-UI/issues/2962
and seems that the author will integrate it in the future. Now I just check uniqueness like that:
```javascript
$myForm.form({
    ...
    onSuccess: function () {
            checkUniquenessAsync(function (isUnque) {
                    if (!isUnique) {
                        $myForm.form('add error', [ 'not unique!!!' ]) 
                    } else {
                        // submit the data
                    }
            });
    }
});
```

**Comment by IonutBajescu** [comment]
Thank you @xiechao06, didn't find issue post when I searched for previous pull requests. 

I'm using a similar approach until we get a bult-in async implementation.
!()[https://i.imgur.com/ZDM4T8V.png]

**Comment by jlukic** [comment]
I'll be considering async in form validations sometime shortly

**Comment by avalanche1** [comment]
+1

**Comment by arist0tl3** [comment]
+1

**Comment by avalanche1** [comment]
I don't think this should be closed, really.

**Comment by IonutBajescu** [comment]
I only closed this because it's a duplicate of #2962 which already has a pull-request submitted #2961.

Sorry for forgetting to leave a note with the reason for closing it.

**Comment by avalanche1** [comment]
I'm sorry, I didn't see you were the OP. :smile:
